<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>InteractiveWidget</class>
 <widget class="QWidget" name="InteractiveWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>703</width>
    <height>430</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout" stretch="1,0">
   <item>
    <layout class="QVBoxLayout" name="verticalLayout_4" stretch="0,1">
     <item>
      <widget class="QLabel" name="label_11">
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="styleSheet">
        <string notr="true">border: 2px dashed rgb(255, 0, 0);
background-color: rgb(218, 212, 44);
color: black;</string>
       </property>
       <property name="text">
        <string>交互提示：
鼠标左键拖动模型旋转，鼠标中键拖动模型移动.
鼠标右键拖动相机转动，键盘WASD控制相机移动.</string>
       </property>
       <property name="margin">
        <number>4</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="PreviewPane" name="previewPane" native="true"/>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QVBoxLayout" name="verticalLayout" stretch="1,1,0">
     <item>
      <widget class="QGroupBox" name="groupBox">
       <property name="title">
        <string>三维静态模型变换</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_2" stretch="1,1,1,2">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_4" stretch="0,0">
          <item>
           <widget class="QLabel" name="label">
            <property name="text">
             <string>位置:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="le_model_pos">
            <property name="minimumSize">
             <size>
              <width>200</width>
              <height>0</height>
             </size>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
            <property name="readOnly">
             <bool>true</bool>
            </property>
            <property name="placeholderText">
             <string>x, y, z</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_5" stretch="0,0">
          <item>
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>旋转:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="le_model_rot">
            <property name="minimumSize">
             <size>
              <width>200</width>
              <height>0</height>
             </size>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
            <property name="readOnly">
             <bool>true</bool>
            </property>
            <property name="placeholderText">
             <string>x, y, z</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_6" stretch="0,0">
          <item>
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>缩放:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="le_model_sca">
            <property name="minimumSize">
             <size>
              <width>200</width>
              <height>0</height>
             </size>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
            <property name="readOnly">
             <bool>true</bool>
            </property>
            <property name="placeholderText">
             <string>x, y, z</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="verticalSpacer_2">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_2">
       <property name="title">
        <string>相机变换</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_3" stretch="1,1,3">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_7" stretch="0,0">
          <item>
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>位置:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="le_camera_pos">
            <property name="minimumSize">
             <size>
              <width>200</width>
              <height>0</height>
             </size>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
            <property name="readOnly">
             <bool>true</bool>
            </property>
            <property name="placeholderText">
             <string>x, y, z</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_8" stretch="0,0">
          <item>
           <widget class="QLabel" name="label_9">
            <property name="text">
             <string>旋转:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="le_camera_rot">
            <property name="minimumSize">
             <size>
              <width>200</width>
              <height>0</height>
             </size>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
            <property name="readOnly">
             <bool>true</bool>
            </property>
            <property name="placeholderText">
             <string>x, y, z</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <widget class="QPushButton" name="bt_reset">
         <property name="text">
          <string>重置变换</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="bt_save">
         <property name="text">
          <string>保存变换</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PreviewPane</class>
   <extends>QWidget</extends>
   <header>gui/uicomponent/previewpane.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
